dev/validate_data_file.py

```diff
@@ -0,0 +1,359 @@
+                                      f'must end in a version number (e.g. "_v1") where the number '
+                                      f'is an integer.')
+
+            validate_model(model_value, header)
+        else:
+            raise ValidationError(f'{header}a model must be either a str or a (YAML) dict (is a '
+                                  f'{type(model_value)})')
+
+
+def validate_model(model: dict, header: str) -> None:
+    """
+    Validates the fifth level of the data file, the dictionary inside a key representing a
+    particular model.
+
+    Parameters
+    ----------
+    model
+        The dictionary containing the data inside a key representing a particular model inside the
+        YAML data file.
+    header
+        The header to prepend to error messages.
+    """
+    if not 'function' in model:
+        raise ValidationError(f'{header}the "function" key is missing.')
+    elif not isinstance(model['function'], str):
+        raise ValidationError(f'{header}the "function" must be a str (is a '
+                              f'{type(model["function"])})')
+
+    if not 'citation' in model:
+        raise ValidationError(f'{header}the "citation" key is missing.')
+    elif not isinstance(model['citation'], list):
+        raise ValidationError(f'{header}the "citation" must be a list of str (is a '
+                              f'{type(model["citation"])})')
+    else:
+        for value in model['citation']:
+            if not isinstance(value, str):
+                raise ValidationError(f'{header}the "citation" must be a list of str (entry '
+                                      f'"{value}" is a {type(model["citation"])})')
+
+    if not 'parameters' in model:
+        raise ValidationError(f'{header}the "parameters" key is missing.')
+    elif not isinstance(model['parameters'], dict):
+        raise ValidationError(f'{header}the "parameters" must be a (YAML) dict (is a '
+                              f'{type(model["parameters"])})')
+
+    if not 'configurations' in model:
+        raise ValidationError(f'{header}the "configurations" key is missing.')
+    elif not isinstance(model['configurations'], dict):
+        raise ValidationError(f'{header}the "configurations" must be a (YAML) dict (is a '
+                              f'{type(model["configurations"])})')
+
+    validate_configurations_dict(model['configurations'], header)
+
+
+def validate_configurations_dict(configurations_dict: dict, header: str) -> None:
+    """
+    Validates the sixth level of the data file, the dictionary inside the ``configurations`` key.
+
+    Parameters
+    ----------
+    configurations_dict
+        The dictionary containing the data inside the ``configurations`` key of the YAML data file.
+    header
+        The header to prepend to the error messages.
+    """
+    if not configurations_dict:
+        raise ValidationError(f'{header}No configurations provided (dictionary is empty)')
```

Comment:
Why should this be an error? It currently fails for VISION which can't be reconfigured.

Reply:
You are right - that should not fail!